### DISCUSSION TITLE
mongoose v6 are no longer connect with `useNewUrlParser` and `useUnifiedTopology`

### DISCUSSION THREAD

**Discussion by chuanxie** [typegoose/mongodb-memory-server#551] [Q&A]
## Versions

- mongodb-memory-server-*: 7.4.2
- mongodb: 4.1.2
- mongoose: 6.0.8
- system: 

Type `{ useNewUrlParser: boolean; useUnifiedTopology: boolean; }` has no properties in common with type `ConnectOptions`
so how to connect to mongodb-memory-server? always timeout

**Comment by hasezoey** [comment]
TL;DR: this is not a problem with mongodb-memory-server

Long: you have this error because mongodb (driver) 4.0 was a re-write, which made such options "default" and no longer be able to be set, which happens when you use mongoose 6.0, which is based on mongodb (driver) 4.1, for more information please read [monoose's migration guide](https://mongoosejs.com/docs/migrating_to_6.html)
